docs/decisions/0002-remove-use-jwt-cookie-header.rst

```diff
@@ -21,14 +21,16 @@ Use of this header has several problems:
 Decision
 --------
 
-Replace the `HTTP_USE_JWT_COOKIE` header with forgiving authentication when using JWT cookies. By "forgiving", we mean that JWT authentication would no longer raise exceptions for failed authentication when using JWT cookies, but instead would simply return None.
+Replace the ``HTTP_USE_JWT_COOKIE`` header with forgiving authentication when using JWT cookies. By "forgiving", we mean that JWT authentication would no longer raise exceptions for failed authentication when using JWT cookies, but instead would simply return None.
 
 By returning None from JwtAuthentication, rather than raising an authentication failure, we enable services to move on to other classes, like SessionAuthentication, rather than aborting the authentication process. Failure messages could still be surfaced using `set_custom_metric` for debugging purposes.
```

Comment:
Is it worth adding that this is how DRF expects its auth classes to behave?

Reply:
Here are the DRF docs for custom authentication: https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication

> To implement a custom authentication scheme, subclass `BaseAuthentication` and override the `.authenticate(self, request)` method. The method should return a two-tuple of `(user, auth)` if authentication succeeds, or `None` otherwise.
>
> In some circumstances instead of returning None, you may want to raise an `AuthenticationFailed` exception from the .authenticate() method.

The start of their custom authentication docs make it seem like you should return None by default, but maybe raise an exception under certain circumstances.

However, I was more familiar with the next statements, which practically state the reverse:

> Typically the approach you should take is:
> - If authentication is not attempted, return `None`. Any other authentication schemes also in use will still be checked.
> - If authentication is attempted but fails, raise an `AuthenticationFailed` exception. An error response will be returned immediately, regardless of any permissions checks, and without checking any other authentication schemes.

This latter section is what I took to be their actual recommendation. Since we actually do check and fail first, we are somewhat going against this DRF "typical approach" by returning None anyway and allowing other authentication methods to be tried.

Given this context, if you still have a recommended change, do you mind sending your proposed text to clarify? Thanks.

Reply:
Gotcha, yea, that does make it confusing.  I think given the confusion in the DRF text, what you have now is good.